edb/schema/objects.py

```diff
@@ -382,20 +387,18 @@ def required(self) -> bool:
     def is_schema_field(self) -> bool:
         return True
 
-    # Breaking Liskov Substitution Principle
-    @overload  # type: ignore
-    def __get__(self, instance: None, owner: Type) -> SchemaField[Type_T]:
-        ...
-
-    @overload  # NoQA: F811
-    def __get__(self, instance: T, owner: Type[T]) -> Type_T:  # NoQA: F811
-        ...
-
-    def __get__(self, instance, owner):  # NoQA: F811
+    def __get__(
+        self,
+        instance: Optional[Object],
+        owner: Type[Object],
+    ) -> Optional[T]:
         if instance is not None:
             raise FieldValueNotFoundError(self.name)
         else:
-            return self
+            raise AttributeError(
+                f"type object {owner.__name__!r} "
```

Comment:
nit: I prefer leading whitespace in such situations since if you do this consistently, it's easier to spot a missing space.